sky/jobs/controller.py

```diff
@@ -474,6 +487,11 @@ def start(job_id, dag_yaml, retry_until_up):
     controller_process = None
     cancelling = False
     try:
+        if (len(managed_job_state.get_nonterminal_job_ids_by_name(None)) >
+                managed_job_utils.NUM_JOBS_THRESHOLD):
+            raise exceptions.ManagedJobUserCancelledError(
+                'Too many concurrent managed jobs are running. '
+                'Please try again later or cancel some jobs.')
```

Comment:
Hmm, this does not make sense, as users should expect the managed jobs being submitted queued on the controller when there is not enough resources on controller, instead of erroring out. Should we instead changing the task CPU requirement based on the memory on controller?

Reply:
If cpu/memory == 1/4, use 0.25 cpu
if cpu/memory == 1/2, use 0.5 cpu